cluster/crds/secrets.crossplane.io_storeconfigs.yaml

```diff
@@ -106,11 +106,6 @@ spec:
                         type: object
                       source:
                         description: Source of the credentials.
-                        enum:
-                        - None
-                        - Secret
-                        - Environment
-                        - Filesystem
```

Comment:
Hrm, this is unexpected.

Reply:
https://github.com/crossplane/crossplane-runtime/blob/release-0.19/apis/common/v1/connection_details.go#L130

I'm still seeing the kubebuilder tag there.

Reply:
https://book.kubebuilder.io/reference/markers/crd-validation.html

The relevant validation comment directive still seems supported, too.

Reply:
I see plenty of other enums and defaults still working fine. I wonder if the issue here is that this particular type contains is embedded, and from another module? (`spec` is an embedded xpv1.SecretStoreConfig from crossplane-runtime.)

Reply:
I just tried a few different controller-tools versions. It looks like I have to go all the way back to 0.8.0 to get it to include this enum - everything from 0.9.2 and onward has the same behaviour we see here.

Reply:
I don't really know how to fix this, and I would prefer not to keep blocking the release. My suspicion is that it's an issue with controller-tools that we would have to fix there.

How do folks feel about merging this PR despite this issue? I feel comfortable with it - it's not ideal, but it's only a slight loosening of schema validation for an alpha API type.

Reply:
I would be fine with going forward if we only lose `enum` and schema validations for that; however, losing the `default` might result in unexpected behavior changes for those using `publishConnectionDetailsTo` API with Kubernetes. 

This is what I get after installation as default store config, which I believe wouldn't be usable until it is patched with Kubernetes as type:

```
kubectl get storeconfig
NAME      AGE     TYPE   DEFAULT-SCOPE
default   2m32s          upbound-system
```

Reply:
... which we can workaround by explicitly setting it [here](https://github.com/crossplane/crossplane/blob/90c2ada429a909a1a49b4ea3374b5a760f3f63f9/internal/initializer/storeconfig.go#L57).
and for all providers once/if we consume this runtime, e.g. [here](https://github.com/crossplane-contrib/provider-aws/blob/bf4088121aa87877cb969c5de3e1b287d6f853ca/cmd/provider/main.go#L111).

Reply:
It looks like this is an issue with controller-runtime only, I tried using a struct from another project with kubebuilder annotations, enums and defaults and it's generating the crd correctly... 🤔

Reply:
replacing https://github.com/crossplane/crossplane/blob/e89ef3c90ccb1bc9f1cb2bd0f2ba3451afbc7332/apis/generate.go#L34 with 
```
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./pkg/v1alpha1/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./pkg/v1beta1/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./pkg/v1/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./apiextensions/v1alpha1/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./apiextensions/v1beta1/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./apiextensions/v1/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./apiextensions/v1/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
//go:generate go run -tags generate sigs.k8s.io/controller-tools/cmd/controller-gen object:headerFile=../hack/boilerplate.go.txt paths=./secrets/... crd:crdVersions=v1 output:artifacts:config=../cluster/crds
```
is solving the issue

Reply:
This works for me - nice catch. I think we're good to go ahead with this workaround.

Reply:
great find on this @phisco! 💪